src/uct/sm/base/sm_iface.c

```diff
@@ -134,16 +98,9 @@ int uct_sm_iface_is_reachable(const uct_iface_h tl_iface,
         return 0;
     }
 
-    if ((ext_addr->super.ext != my_addr.super.ext) ||
-        (ext_addr->super.id  != my_addr.super.id)) {
-            return 0;
-    }
-
-    if (!ext_addr->super.ext) {
-        return 1;
-    }
-
-    return ext_addr->ipc_ns == my_addr.ipc_ns;
+    return (ext_addr->super.id == my_addr.super.id) &&
+           (!(ext_addr->super.id & UCS_SM_IFACE_ADDR_FLAG_EXT) ||
+            (ext_addr->ipc_ns == my_addr.ipc_ns));
```

Comment:
add comment to describe this logic

Reply:
unhandled?

Reply:
I split this evaluation to 3 ```if```` due to clang warning.
Added comment